src/manager.cpp

```diff
@@ -688,21 +700,56 @@ sdbusplus::async::task<bool>
     lg2::debug("Elapsed time for sync: [{DURATION_SECONDS}] seconds for {CMD}",
               "DURATION_SECONDS", syncElapsedTime.count(), "CMD", syncCmd);
 
-    if (ret.first != 0)
+    if (ret.first == 0)
+    {
+        co_return true;
+    }
+
+    const std::string srcLog = srcPath.empty() ? dataSyncCfg._path.string()
```

Comment:
nit: this function is already using auto, e.g. lines 697,698, so probably appropriate here as well.